### PR TITLE
[docs] Change italicized version references to admonitions, add more version information

### PR DESCRIPTION
**Pull request by glennmatthews**
# Closes: #DNE
# What's Changed

- Replace references to Nautobot version differences in the docs from italicized text to admonition blocks, e.g.:

| Before | After |
| -- | -- |
| image | image |
| image | image |
| image | image |

- Additionally, based on a quick review of the release-note "overview" sections, add a bunch of version difference notes that weren't previously present in the docs.

# TODO

- [x] Explanation of Change(s)
- [x] Attached Screenshots, Payload Example
- n/a Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- n/a Example Plugin Updates (when adding/changing features)
- n/a Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/docs/additional-features/jobs.md [hunk 1]
```diff
@@ -132,6 +132,8 @@ A list of strings (field names) representing the order your job [variables](#var
 
 #### `has_sensitive_variables`
 
+!!! check "Added in version 1.3.10"
```
Comment (bryanculver) [inline nautobot/docs/additional-features/jobs.md:135]:
```suggestion
!!! info "Added in version 1.3.10"
```

For any general purpose version admonition should we use info instead?

**Comment by glennmatthews** [comment]
I went with `check` since its banner is green which to me makes more sense for "added/enhanced" than the light blue of `info`, plus we already use `info` and the similarly-colored `tip` for a lot of other purposes and I wanted it to stand out a bit. Willing to change it if desired, just explaining my thought process. :-) 

Full list of default admonition types: https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types

There's also the option to add a [custom admonition type](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#custom-admonitions) if we want to.

**Comment by glennmatthews** [comment]
Screen Shot 2022-09-12 at 2 03 54 PM

**Comment by smk4664** [comment]
If Nautobot Apps adopt the same admonitions, then making this a stand-alone mkdocs plugin would be preferable.

**Comment by glennmatthews** [comment]
Published [mkdocs-version-annotations](https://pypi.org/project/mkdocs-version-annotations/) plugin and updated this to use it. The annotations look especially nice in dark mode:

image

File: changes/2400.changed [hunk 2]
```diff
@@ -0,0 +1 @@
+Improved formatting of version changes in the documentation
```
Comment (gsnider2195) [inline changes/2400.changed:1]:
```suggestion
Improved formatting of version changes in the documentation.
```

File: nautobot/docs/administration/nautobot-server.md [hunk 3]
```diff
@@ -129,10 +129,9 @@ nautobot=> \q
 
 ### `dumpdata`
 
-<!-- markdownlint-disable MD036 -->
-_Changed in version 1.3.0: `extras.job` should now be included (removed `--exclude extras.job`)_
-_Changed in version 1.3.0: `django_rq` should now be excluded (add `--exclude django_rq`)_
- <!-- markdownlint-enable MD036 -->
++/- 1.3.0
```
Comment (gsnider2195) [inline nautobot/docs/administration/nautobot-server.md:132]:
```suggestion
+/- 1.3.5
```

https://nautobot.readthedocs.io/en/stable/release-notes/version-1.3/#fixed_5
#1786

Comment (glennmatthews) [reply nautobot/docs/administration/nautobot-server.md:132]: 1.3.5 is where the documentation was updated, but 1.3.0 is the point at which this recommendation became relevant.

File: nautobot/docs/plugins/index.md [hunk 4]
```diff
@@ -9,19 +9,31 @@ The Nautobot plugin architecture allows for plugins to do any or all of the foll
...
-* **Add additional secrets providers.** A plugin can add support for retrieving [Secret](../models/extras/secret.md) values from additional sources or external systems.
-* **Override already-defined views.** _Added in version 1.4.0_: A plugin can define a view which can be set to override a view from the core set of views or another plugin's view.
+
++++ 1.1.0
+    * **Register additional Jinja2 filters.** A plugin can define custom Jinja2 filters to be used in computed fields, webhooks, custom links, and export templates.
+
++++ 1.2.0
+    * **Populate extensibility features in the database.** A plugin can add content to the Nautobot database when installed, such as automatically creating new custom fields, relationships, and so forth.
+
+    * **Add additional secrets providers.** A plugin can add support for retrieving [Secret](../models/extras/secret.md) values from additional sources or external systems.
```
Comment (gsnider2195) [inline nautobot/docs/plugins/index.md:33]: Is this supposed to be under the 1.2.0 heading?

**Review by gsnider2195 — APPROVED** [review]

**Pull request merged**